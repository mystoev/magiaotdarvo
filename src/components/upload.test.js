import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';

import Upload from './upload';

describe('<Upload /> tests', () => {
  it('should render', () => {
    const { container } = render(<Upload />);

    expect(container).toMatchSnapshot();
  });

  it('should upload', async () => {
    const file = new File(['some content'], 'file1.jpg', { type: 'image/jpg' });
    const onUploadMock = jest.fn();
    const { queryByTestId } = render(<Upload onUpload={onUploadMock} />);

    fireEvent.change(queryByTestId('file-input'), { target: { files: [file] } });

    await waitFor(() => {
      expect(onUploadMock).toHaveBeenCalledWith([
        {
          data: 'data:image/jpg;base64,c29tZSBjb250ZW50',
          name: 'file1.jpg'
        }
      ]);
    });
  });
});
