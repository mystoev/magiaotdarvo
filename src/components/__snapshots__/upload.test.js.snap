// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Upload /> tests should render 1`] = `
<div>
  <div
    class="parent"
  >
    <div
      class="file-upload"
    >
      <h3>
        Натисни тук, за да качиш снимки
      </h3>
      <input
        data-testid="file-input"
        multiple=""
        type="file"
      />
    </div>
  </div>
</div>
`;
